apps/antalmanac/src/components/Header/SettingsMenu.tsx

```diff
@@ -1,96 +1,142 @@
-import {
-    Box,
-    Button,
-    FormControl,
-    FormControlLabel,
-    Paper,
-    Popover,
-    Radio,
-    RadioGroup,
-    Typography,
-} from '@material-ui/core';
-import { Settings } from '@material-ui/icons';
-import { ChangeEvent, useEffect, useState } from 'react';
+import { useState } from 'react';
+import { Box, Button, ButtonGroup, Divider, Drawer, IconButton, Typography } from '@material-ui/core';
+import { Close, DarkMode, LightMode, Settings, SettingsBrightness } from '@mui/icons-material';
 
-import { toggleTheme } from '$actions/AppStoreActions';
-import AppStore from '$stores/AppStore';
 import { useTimeFormatStore } from '$stores/TimeStore';
+import { useThemeStore } from '$stores/ThemeStore';
 
-function SettingsMenu() {
-    const [anchorEl, setAnchorEl] = useState<HTMLElement>();
-    const [theme, setTheme] = useState(AppStore.getTheme());
+function ThemeMenu() {
+    const handleThemeChange = (event: React.MouseEvent<HTMLButtonElement>) => {
+        useThemeStore.getState().setTheme(event.currentTarget.value);
```

Comment:
Wait, that's illegal. Never invoke hooks outside of the top level of a function. It might work because Zustand is "hook-like" and not a real hook? But don't do that regardless.

Apply this to the other relevant handlers in the file.

```tsx
function ThemeMenu() {
  const setTheme = useThemeStore((store) => store.setTheme)

   const handleThemeChange = useCallback((event: React.MouseEvent<HTMLButtonElement>) => {
     setTheme(event.currentTarget.value);
  };
}, [])
```

P.S.
I didn't come up with that rule, some brain-rotted, React-andy troglodyte soydev did.